pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -477,7 +477,13 @@ func (p *Pod) Stop(ctx context.Context, c client.Client, _ []podset.PodSetInfo,
 }
 
 func SetupIndexes(ctx context.Context, indexer client.FieldIndexer) error {
-	return jobframework.SetupWorkloadOwnerIndex(ctx, indexer, gvk)
+	if err := SetupPodGroupNameIndex(ctx, indexer); err != nil {
```

Comment:
Nit
```suggestion
	if err := indexer.IndexField(ctx, &corev1.Pod{}, PodGroupNameCacheKey, IndexPodGroupName); err != nil {
```

Prefer less indirection for a one-liner